browser/ui/views/brave_actions/brave_shields_action_view.cc

```diff
@@ -349,5 +357,37 @@ void BraveShieldsActionView::OnTabStripModelChanged(
   }
 }
 
+void BraveShieldsActionView::OnTabGroupChanged(const TabGroupChange& change) {
+  if (change.type != TabGroupChange::kCreated ||
+      change.GetCreateChange()->reason() !=
+          TabGroupChange::TabGroupCreationReason::
+              kInsertedFromAnotherTabstrip) {
+    return;
+  }
+
+  const int active_index = tab_strip_model_->active_index();
+  if (tab_strip_model_->empty() || active_index == TabStripModel::kNoTab) {
+    return;
+  }
+
+  // Why we have to find previous active web contents and reset callback here?
+  // We remove observer when it becomes inactive tab via
+  // OnTabStripModelChanged(). However, it doesn't work as expected when active
+  // tab is changed by tab group re-attaching. When it's re-attached, new tab
+  // from tab group is activated but |selection.old_contents| is null when
```

Comment:
Commented about this at https://chromium-review.googlesource.com/c/chromium/src/+/6283627/comment/3501f1f5_08b2854e/

Reply:
This workaround code also could be removed later - https://github.com/brave/brave-browser/issues/45335